### ISSUE TITLE
Got banned because of nothing

### ISSUE THREAD

**Issue by resatkrlc** [adiwajshing/Baileys#807]
**Describe the bug**
I got banned, but still idk why it happned

What i did?

Really nothing. NO message, no anything. Only using a function when someone is being online or offilne. That's all.

What can i do guys? Thanks

**Comment by LuanRT** [comment]
A ban is something totally unpredictable, sorry but there's nothing we can do about it.
You may want to try ‘humanizing’ your bot's actions a bit tho.

**Comment by resatkrlc** [comment]
> A ban is something totally unpredictable, sorry but there's nothing we can do about it. You may want to try ‘humanizing’ your bot's actions a bit tho.

Yes you're right but i still wonder, do you think i got banned because of using emulator?

and i've  been just checking their status. Like if they are online or offline

**Comment by FaizBastomi** [comment]
It's possible that WhatsApp detects you are using an emulator. And on the original WhatsApp webpage it says that they don't support emulators. The possibility of BAN in the emulator is also greater.

**Comment by BlackAmda** [comment]
That's not belong to baileys as a bug. 🤷🏻‍♂️???

**Comment by canove** [comment]
It may be due to Baileys. I had a few bans too. The reason is not clear yet.

**Comment by SheIITear** [comment]
> It may be due to Baileys. I had a few bans too. The reason is not clear yet.

Obviously. You're breaking their ToS, the issue creator breaks it twice by using an emulator. There's no guaranteed way to prevent getting banned.

**Comment by canove** [comment]
Sure, we need to keep up the investigation. @Ruun are doing some research on it in Discord. His theory is 

> that has some "w:stats" events that keeps sending telemetry about browser, version, user activity and etc, this is probably being used to ban if you don't send

I've been using Baileys Legacy for a long time with no bans, hope we can get the same with new Baileys MD.

**Comment by ilyaborodin** [comment]
We bought a new sim card and scanned with it qr. Made few small test and got banned. I added video with all actions that we did. In left column all messages except first "oi" were send via Baileys, "oi"was sent from wa business app. 
Qr scanned by wa business app.
I tested a lot with my phone number locally and did not get banned. 
So the reason in my mind:
1) i did not read messages before send a new message
2) this is related to new wa user
3) server is located in USA, but the phone is in Brazil (but with bailey legacy for all the time there were no problems with this)
4) or smth with library

I will fix 1, 3 and test again

https://user-images.githubusercontent.com/48887908/141800544-5c351f0c-7a36-4f35-9baf-f1992ce72aac.MP4

**Comment by canove** [comment]
Is there a way to send these button messages using just WhatsappWeb and not Baileys?

If it doesn't, it could be that this usage is being detected as inappropriate and causing the blocks...

Can you test it again, without button / list messages?

**Comment by ilyaborodin** [comment]
@canove 
So, my reason was in server ip. Server was located in usa(Before i used it for baileys legacy staging). When i scanned and tried to anything(send any message or create group) i got banned. I only tried new sim cards and virtual sim cards for test. Also i tried to use deferent vpns on my phone to move my location to usa, but this was unsuccessful. But locally this sim cards and virtuals numbers with the same actions worked good. I made the tests locally (create ± 20 groups in a row, send a lot of messages in one chat), but did not get banned. So i created a new server ec2(sa-east-1). And it works fine now. I created ±20 groups, send messages. Everything good. I tested with the same simcards from br and phone from br and i tested with virtual card and phone from Europe. So i think ip of USA server was in "Blacklist" for some reasons.
Or maybe old provider modify headers or smth like this. I used in usa not aws
